pkg/generator/helper/helper.go

```diff
@@ -364,27 +430,34 @@ func ProcessMetaFiles(metadataFiles []string, targetDir string, stepHelperData S
 		checkError(err)
 
 		step := stepTemplate(myStepInfo, "step", stepGoTemplate)
-		err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v_generated.go", stepData.Metadata.Name)), step, 0644)
+		err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v_generated.go", stepName)), step, 0644)
 		checkError(err)
 
 		test := stepTemplate(myStepInfo, "stepTest", stepTestGoTemplate)
-		err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v_generated_test.go", stepData.Metadata.Name)), test, 0644)
+		err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v_generated_test.go", stepName)), test, 0644)
 		checkError(err)
 
-		exists, _ := piperutils.FileExists(filepath.Join(targetDir, fmt.Sprintf("%v.go", stepData.Metadata.Name)))
+		exists, _ := piperutils.FileExists(filepath.Join(targetDir, fmt.Sprintf("%v.go", stepName)))
 		if !exists {
 			impl := stepImplementation(myStepInfo, "impl", stepGoImplementationTemplate)
-			err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v.go", stepData.Metadata.Name)), impl, 0644)
+			err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v.go", stepName)), impl, 0644)
 			checkError(err)
 		}
 
-		exists, _ = piperutils.FileExists(filepath.Join(targetDir, fmt.Sprintf("%v_test.go", stepData.Metadata.Name)))
+		exists, _ = piperutils.FileExists(filepath.Join(targetDir, fmt.Sprintf("%v_test.go", stepName)))
 		if !exists {
 			impl := stepImplementation(myStepInfo, "implTest", stepGoImplementationTestTemplate)
-			err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v_test.go", stepData.Metadata.Name)), impl, 0644)
+			err = stepHelperData.WriteFile(filepath.Join(targetDir, fmt.Sprintf("%v_test.go", stepName)), impl, 0644)
 			checkError(err)
 		}
 	}
+	// expose metadata functions
+	exists, _ := piperutils.FileExists(filepath.Join(targetDir, metadataGeneratedTemplate))
+	if !exists {
```

Comment:
This looks wrong to me. If you run "go generate", it should overwrite and update this file.

Reply:
good catch!

Reply:
thanks!